src/www/ui/api/Controllers/UserController.php

```diff
@@ -60,6 +60,115 @@ public function getUsers($request, $response, $args)
     return $response->withJson($users, 200);
   }
 
+  /**
+   * Create a user
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function addUser($request, $response, $args)
+  {
+    $dbManager = $this->dbHelper->getDbManager();
```

Comment:
This function can be simplified heavily by importing the following, "existing" function instead of copying the content of the function:

https://github.com/fossology/fossology/blob/011573743304b3a56a858fcbb77083c4eebcdfd1/src/www/ui/user-add.php#L44-L141

Reply:
There might be some changes required in `user-add.php` to handle outside requests.

1. The base class has to be changed from `FO_Plugin` to `DefaultPlugin`.
2. Function `Output()` has to be changed to `function handle(Request $request)`.
3. The function `Add()` will now be `public` and accept `Request` object from `handle()`.

For similar implementation, see https://github.com/fossology/fossology/blob/master/src/www/ui/user-edit.php

Reply:
Changing base class to `DefaultPlugin` seems to cause error in the application.